src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -668,15 +693,39 @@ private static DkmEvaluationResult CreateEvaluationResultFromException(Exception
                 DataItem: null);
         }
 
-        internal static void ContinueWithExceptionHandling(CompletionRoutine onCompleted, CompletionRoutine<Exception> onException)
+        private sealed class WorkList
```

Comment:
Would it make sense for WorkList to be IDisposable and assert that _completionRoutine is null (or, if not Dispose, then some other sort of validation that Execute always gets called when it needs to)?  It worries me a little that we might forget to call Execute on a WorkList that has a continuation.